src/srs.jl

```diff
@@ -0,0 +1,174 @@
+struct KeyEntry
+    keyid::UInt16
+    tifftaglocation::UInt16
+    count::UInt16
+    value_offset::UInt16
+end
+
+struct GeoKeys
+    key_directory_version::UInt16
+    key_reversion::UInt16
+    minor_revision::UInt16
+    number_of_keys::UInt16
+    keys::Vector{KeyEntry}
+end
+
+# version numbers are fixed in the LAS specification
+GeoKeys(keys::Vector{KeyEntry}) = GeoKeys(0x0001, 0x0001, 0x0000, UInt16(length(keys)), keys)
+
+struct GeoDoubleParamsTag
+    double_params::Vector{Float64}
+end
+
+struct GeoAsciiParamsTag
+    ascii_params::String
+    nb::Int  # number of bytes
+    GeoAsciiParamsTag(s::AbstractString, nb::Integer) = new(ascii(s), Int(nb))
+end
+
+const id_geokeydirectorytag = UInt16(34735)
+const id_geodoubleparamstag = UInt16(34736)
+const id_geoasciiparamstag = UInt16(34737)
+
+"test whether a vlr is a GeoKeyDirectoryTag, GeoDoubleParamsTag or GeoAsciiParamsTag"
+is_srs(vlr::LasVariableLengthRecord) = vlr.record_id in (
+    id_geokeydirectorytag,
+    id_geodoubleparamstag,
+    id_geoasciiparamstag)
+
+# length in number of bytes
+Base.length(data::GeoKeys) = 8 * Int(data.number_of_keys) + 8
+Base.length(data::GeoDoubleParamsTag) = length(data.double_params) * 8
+Base.length(data::GeoAsciiParamsTag) = data.nb
+
+"Construct a projection VLR based on an EPSG code"
+function LasVariableLengthRecord(header::LasHeader, epsg::Integer)
+    if is_wkt(header)
+        throw(ArgumentError("Setting the SRS in WKT format not implemented"))
```

Comment:
Perhaps move this error check out of the VLR construction and next to the place where the VLR is `push!`ed onto the list of VLRs for the header?

The fact that header is required to allocate the VLR seems like unnecessary coupling.